apps/server/src/modules/files/uc/sync-embedded-files.uc.ts

```diff
@@ -12,86 +12,114 @@ import { SyncFilesStorageService } from './sync-files-storage.service';
+				if (item.content === undefined || !('text' in item.content)) {
+					return;
+				}
 
-			const regEx = new RegExp(`(?<=src=${fileUrlRegex}).+?(?=&amp;)`, 'g');
-			const contentFileIds = item.content.text.match(regEx);
+				const contentFileIds = this.extractFileIdsFromContent(item.content.text);
+
+				if (contentFileIds !== null) {
+					fileIds.push(...contentFileIds);
+				}
+			});
+		}
+		if (entity instanceof Task) {
+			const contentFileIds = this.extractFileIdsFromContent(entity.description);
 
 			if (contentFileIds !== null) {
-				lessonFileIds.push(...contentFileIds);
+				fileIds.push(...contentFileIds);
 			}
-		});
+		}
 
-		return _.uniq(lessonFileIds).map((id) => new ObjectId(id));
+		return _.uniq(fileIds).map((id) => new ObjectId(id));
 	}
 
-	private async syncFiles(files: SyncFileItem[]) {
-		// eslint-disable-next-line no-await-in-loop
-		const promises = files.map((file) => {
-			return this.sync(file);
-		});
+	private extractFileIdsFromContent(text: string) {
+		const regEx = new RegExp(`(?<=src=${fileUrlRegex}).+?(?=&amp;)`, 'g');
+		const contentFileIds = text.match(regEx);
+
+		return contentFileIds;
+	}
+
+	private async syncFiles(files: SyncFileItem[], entity: Lesson | Task) {
+		const promises = files.map((file) => this.sync(file, entity));
 		await Promise.all(promises);
 	}
 
-	private async sync(file: SyncFileItem) {
+	private async sync(file: SyncFileItem, entity: Lesson | Task) {
 		try {
 			await this.syncFilesMetaDataService.prepareMetaData(file);
 			await this.syncFilesStorageService.syncS3File(file);
 			await this.syncFilesMetaDataService.persistMetaData(file);
-			await this.updateLessonsLinks(file);
+			await this.updateEntityLinks(file, entity);
 			this.logger.log(`Synced file ${file.source.id}`);
 		} catch (error) {
 			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 			const stack: string = 'stack' in error ? (error as Error).stack : error;
 			this.logger.error(file, stack);
+			this.wrongIds.push(entity._id);
 			await this.syncFilesMetaDataService.persistError(file, stack);
 		}
 	}
 
-	private async updateLessonsLinks(file: SyncFileItem) {
-		const lesson = await this.embeddedFilesRepo.findLesson(new ObjectId(file.parentId));
-
-		if (lesson) {
-			lesson.contents = lesson.contents.map((item: IComponentProperties) => {
-				const regex = new RegExp(`${fileUrlRegex}${file.source.id}.+?"`, 'g');
-				const newUrl = `"/api/v3/file/download/${file.fileRecord.id}/${file.fileRecord.name}"`;
-
+	private async updateEntityLinks(file: SyncFileItem, entity: Task | Lesson, errorUrl?: string) {
+		if (file.parentType === FileRecordParentType.Lesson && entity instanceof Lesson) {
+			entity.contents = entity.contents.map((item: IComponentProperties) => {
 				if ('text' in item.content) {
-					item.content.text = item.content.text.replace(regex, newUrl);
+					item.content.text = this.replaceLink(item.content.text, file, errorUrl);
```

Comment:
Spannend, dass das hier funktioniert. Rein vom Code her hätte ich gedacht, dass wenn zwei Files in dem gleichen Text sind, dass nur eines davon migiriert wird. Aber ORM scheint hier irgendwie magic zu machen. Können wir da noch mal drüber sprechen?